frontend/src/hooks/useListFilter.ts

```diff
@@ -10,18 +11,22 @@ export const useListFilter = ({
   const history = useHistory();
   const searchParams = new URLSearchParams(location.search);
 
-  const setFilter = (key: string, value: string) => {
-    if (value || setEmptyValues) {
-      searchParams.set(key, value);
-    } else {
-      searchParams.delete(key);
-    }
-    // Reset pagination if filters change
-    if (key !== 'offset') {
-      searchParams.delete('offset');
-    }
-    history.push({ search: searchParams.toString() });
-  };
+  const setFilter = useCallback(
+    (key: string, value: string) => {
+      if (value || setEmptyValues) {
+        searchParams.set(key, value);
+      } else {
+        searchParams.delete(key);
+      }
+      // Reset pagination if filters change
+      if (key !== 'offset') {
+        searchParams.delete('offset');
+      }
+      history.push({ search: searchParams.toString() });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [setEmptyValues]
+  );
```

Comment:
This creates very weird issues.

useCallback is about memoization, but the way you use it here without passing `searchParams` causes the method `setFilter` to not be updated when searchParams changes. Weirdly as long as only this `setFilter` method changes the `searchParams` then the newly updated `searchParams`and the `searchParams` memoized stay in sync and all works well.

It starts breaking when search params are updated somewhere else, for example by navigating back or by opening a link that does not specify the language filter.

Example to reproduce:

1. Home at http://localhost:3000/ 
2. Select Recommendations
3. Open filters and select "All Time"
4. Navigate back
5. Observe that the filter is back to "A month ago" as expected
6. Select max duration 20
7. /!\ Observe that the data filter is set to "All Time" but should have stayed "A month ago"

Other issue:

1. Home at http://localhost:3000/ 
2. Select Recommendations
4. Select max duration 20
5. Navigate back
6. Observe that the filtering and url correctly do not have any max duration, but the TextInput still shows max duration at 20

Reply:
I see, it's more complex than I expected.

I'll give it a second look after the research PR.

Reply:
I still need to do more tests to be sure I didn't break anything else, but the two issues mentioned above should be fixed by: https://github.com/tournesol-app/tournesol/pull/1029/commits/4581aa76da0025847adcbdbba8a5202bb218df68